src/libxively/sft/xi_sft_logic.c

```diff
@@ -60,6 +60,10 @@ xi_state_t xi_sft_free_context( xi_sft_context_t** context )
 {
     if ( NULL != context && NULL != *context )
     {
+        xi_sft_on_message_file_chunk_checksum_final( *context );
+
+        xi_bsp_io_fs_close( ( *context )->update_file_handle );
```

Comment:
Wonder if we should pass an error to this function in case someone wants to roll back the file, if possible.

Reply:
? You mean "error, file close requested since there was an error (socket disconnection) during file content processing" or something like this?
And here the application may continue download from this point?
If this is the idea, I'd say this is too advanced subfeature for our current status: undestaffed + this SFT feature isn't sold at all yet. And nobody requested this feature.

Reply:
I wouldn't say "download again later" but perhaps "Not commit this file change, or Roll it back"
I could see some systems that don't overwrite files, but write to a temporary file until the checksum calculation occurs, and then moves the file.